### PR TITLE
[ADD] doc: Mobile JavaScript documentation

### PR DESCRIPTION
**Pull request by dpr-odoo**
Description of the issue/feature this PR addresses:

New Mobile JavaScript Documentation for Developer

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: doc/reference/mobile.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+.. _reference/mobile:
+
+==================
+Mobile JavaScript
+==================
+
+Introduction
+============
+
+In Odoo 10.0 We released new mobile app, allows you to access all **Odoo apps** (Even your customized modules).
```
Comment (est-odoo) [inline doc/reference/mobile.rst]: In Odoo 10.0 **we** released **a** new mobile app

File: doc/reference/mobile.rst [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+==================
+Mobile JavaScript
+==================
+
+Introduction
+============
+
+In Odoo 10.0 We released new mobile app, allows you to access all **Odoo apps** (Even your customized modules). 
+
+The application is combination of **Odoo Web** and **Native Mobile components**, In other words it is a Odoo Web instance loaded inside native WebView container of mobile.
```
Comment (est-odoo) [inline doc/reference/mobile.rst]: The application is **a** combination  [...]  **in** other words

File: doc/reference/mobile.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+Mobile rpc object provide list of methods available (Works only with Mobile App). 
+
+Just check for method availability and execute it.
+
+Methods
+-------
+
+.. note:: Each of the methods returns JQuery Deffered object with returning data JSON dictionary
+
+``showToast(args)``
```
Comment (xmo-odoo) [inline doc/reference/mobile.rst]: Use ``.. js::function:` sphinx directive to document JS functions: http://www.sphinx-doc.org/en/stable/domains.html#the-javascript-domain

File: doc/reference/mobile.rst [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+``showSnackBar(args)``
+......................
+
+Snackbars provide lightweight feedback about an operation. They show a brief message at the bottom of the screen on mobile and lower left on larger devices. Snackbars appear above all other elements on screen and only one can be displayed at a time.
+
+.. code-block:: javascript
+
+	mobile.methods.showSnackBar({'message': 'Message is deleted', 'btn_text': 'Undo'}).then(function(result){
+		if(result == true){
```
Comment (xmo-odoo) [inline doc/reference/mobile.rst]: Also known as `if (result)`

File: doc/reference/mobile.rst [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+``scanBarcode()``
+.................
+
+The Barcode API detects barcodes in real-time, on device, in any orientation.
+
+It reads the following barcode formats:
+
+* 1D barcodes: EAN-13, EAN-8, UPC-A, UPC-E, Code-39, Code-93, Code-128, ITF, Codabar
+* 2D barcodes: QR Code, Data Matrix, PDF-417, AZTEC
```
Comment (xmo-odoo) [inline doc/reference/mobile.rst]: Is it really work making a distinction between 1D and 2D barcodes?

Comment (dpr-odoo) [inline doc/reference/mobile.rst]: We are using Google's vision api, so we just taken supported types from documentation. https://developers.google.com/vision/android/barcodes-overview

**Comment by dpr-odoo** [comment]
@xmo-odoo Most of the images are output in android device. And one image for explaining method structure. (how to use it) 190 KB and 300 KB banner.

File: doc/reference/mobile.rst (hunk 5 above)
Comment (Yenthe666) [inline doc/reference/mobile.rst]: @xmo-odoo it is actually nice that this is specified, it doesn't hurt anyone to have it split.

**Pull request merged**